### DISCUSSION TITLE
Vault - Extracting choice data

### DISCUSSION THREAD

**Discussion by stephenwf** [digirati-co-uk/iiif-canvas-panel#25] [Hyperion demos]
This demo uses "Vault" api to load in a presentation 2 or 3 manifest and traverse through the canvas, annotation page, annotation and finally down the body. The body is a choice. Hyperion does not dereference each of the choices, but maybe it does need to do this.

----

## [Demo](https://3nqwz.csb.app/) | [Interactive sandbox](https://codesandbox.io/s/manifest-with-choices-3nqwz)

<a href="https://3nqwz.csb.app/">
<img src="https://user-images.githubusercontent.com/8266711/128694076-17948cb2-77c7-4cef-8549-e8aab3731810.png" />
</a>

**Comment by stephenwf** [comment]
Hyperion _could_ extract each of the images:
```json
{
  "id": "2eba27a0",
  "type": "Choice",
  "items": [
    {
      "id": "https://iiif.io/api/image/3.0/example/reference/421e65be2ce95439b3ad6ef1f2ab87a9-dee-natural/full/max/0/default.jpg",
      "type": "ContentResource"
    },
    {
      "id": "https://iiif.io/api/image/3.0/example/reference/421e65be2ce95439b3ad6ef1f2ab87a9-dee-xray/full/max/0/default.jpg",
      "type": "ContentResource"
    }
  ]
}
```

but the problem is that choices have labels that are properties of the choice (e.g. "Natural light" doesn't label the resource on it's own). 

An alternative may be:
```json
{
  "id": "2eba27a0",
  "type": "Choice",
  "items": [
    {
      "id": "https://iiif.io/api/image/3.0/example/reference/421e65be2ce95439b3ad6ef1f2ab87a9-dee-natural/full/max/0/default.jpg",
      "label": {"en": ["Natural light"]},
      "type": "ContentResource"
    },
    {
      "id": "https://iiif.io/api/image/3.0/example/reference/421e65be2ce95439b3ad6ef1f2ab87a9-dee-xray/full/max/0/default.jpg",
      "label": {"en": ["X-Ray"]},
      "type": "ContentResource"
    }
  ]
}
```

**Reply by tomcrane** [reply]
Bug (?) `body` of the Choice anno is reported as type: ContentResource, but it's a `Choice` when resolved.

![image](https://user-images.githubusercontent.com/1443575/128860760-beb6b580-615a-4da1-bcde-d4e762e3a682.png)

You'd need to see it was a `Choice` before this de-ref?

**Reply by stephenwf** [reply]
Yes possibly makes sense to keep choices inline and not dereference them

**Reply by stephenwf** [reply]
Related behaviour / bug: https://github.com/digirati-co-uk/iiif-canvas-panel/discussions/35

**Comment by tomcrane** [comment]
(More generally, should _everything_ include its label in the ref that comes back from Vault? You often need the labels on things, to render in UI, and this might be before fetching the full object from the Vault). See similar discussion on (link tbc).
